x-pack/plugins/lens/public/editor_frame_plugin/editor_frame/editor_frame.test.tsx

```diff
@@ -439,4 +421,284 @@ describe('editor_frame', () => {
+        <EditorFrame
+          visualizationMap={{
+            testVis: {
+              ...mockVisualization,
+              getSuggestions: () => [
+                {
+                  datasourceSuggestionId: 0,
+                  score: 0.5,
+                  state: {},
+                  title: 'Suggestion2',
+                },
+                {
+                  datasourceSuggestionId: 0,
+                  score: 0.8,
+                  state: {},
+                  title: 'Suggestion1',
+                },
+              ],
+            },
+            testVis2: {
+              ...mockVisualization,
+              getSuggestions: () => [
+                {
+                  datasourceSuggestionId: 0,
+                  score: 0.4,
+                  state: {},
+                  title: 'Suggestion4',
+                },
+                {
+                  datasourceSuggestionId: 0,
+                  score: 0.45,
+                  state: {},
+                  title: 'Suggestion3',
+                },
+              ],
+            },
+          }}
+          datasourceMap={{
+            testDatasource: {
+              ...mockDatasource,
+              getDatasourceSuggestionsFromCurrentState: () => [{ state: {}, tableColumns: [] }],
+            },
+          }}
+          initialDatasourceId="testDatasource"
+          initialVisualizationId="testVis"
+        />
+      );
+
+      await waitForPromises();
+
+      // TODO why is this necessary?
+      instance.update();
+      const suggestions = instance.find('[data-test-subj="suggestion"]');
+      expect(suggestions.map(el => el.text())).toEqual([
+        'Suggestion1',
+        'Suggestion2',
+        'Suggestion3',
+        'Suggestion4',
+      ]);
+    });
+
+    it('should switch to suggested visualization', async () => {
+      const newDatasourceState = {};
+      const suggestionVisState = {};
+      const instance = mount(
+        <EditorFrame
+          visualizationMap={{
+            testVis: {
+              ...mockVisualization,
+              getSuggestions: () => [
+                {
+                  datasourceSuggestionId: 0,
+                  score: 0.8,
+                  state: suggestionVisState,
+                  title: 'Suggestion1',
+                },
+              ],
+            },
+            testVis2: mockVisualization2,
+          }}
+          datasourceMap={{
+            testDatasource: {
+              ...mockDatasource,
+              getDatasourceSuggestionsFromCurrentState: () => [
+                { state: newDatasourceState, tableColumns: [] },
+              ],
+            },
+          }}
+          initialDatasourceId="testDatasource"
+          initialVisualizationId="testVis2"
+        />
+      );
+
+      await waitForPromises();
+
+      // TODO why is this necessary?
+      instance.update();
+
+      act(() => {
+        instance.find('[data-test-subj="suggestion"]').simulate('click');
+      });
```

Comment:
Why does this need an `act()` call? I thought enzyme was smart enough to know how to trigger these changes with `simulate`

Reply:
It should, but it currently doesn't - this is probably fixed when we upgrade enzyme and enzyme react adapter (should also get rid of the "you did something outside of act" error which get logged to console while running the tests currently)